src/c-stdaux-generic.h

```diff
@@ -134,22 +134,25 @@ extern "C" {
 #define _c_boolean_expr_(_x) _c_internal_boolean_expr_(__COUNTER__, _x)
 #if defined(C_COMPILER_GNUC)
 #  define _c_internal_boolean_expr_(_uniq, _x)                                  \
-        __extension__ ({                                                        \
-                int C_VAR(b, _uniq);                                            \
+        (__builtin_constant_p(_x)                                               \
+                ? (!!(_x))                                                      \
+                : __extension__ ({                                              \
+                        int C_VAR(b, _uniq);                                    \
                                                                                 \
-                /*                                                              \
-                 * Avoid any extra parentheses around the evaluation of `_x` to \
-                 * allow `-Wparentheses` to warn about use of `x = ...` and     \
-                 * instead suggest `(x = ...)` or `x == ...`.                   \
-                 */                                                             \
+                        /*                                                      \
+                         * Avoid any extra parentheses around the evaluation of \
+                         * `_x` to allow `-Wparentheses` to warn about use of   \
+                         * `x = ...` and instead suggest `(x = ...)` or         \
+                         * `x == ...`.                                          \
+                         */                                                     \
                                                                                 \
-                if (_x)                                                         \
-                        C_VAR(b, _uniq) = 1;                                    \
-                else                                                            \
-                        C_VAR(b, _uniq) = 0;                                    \
+                        if (_x)                                                 \
+                                C_VAR(b, _uniq) = 1;                            \
+                        else                                                    \
+                                C_VAR(b, _uniq) = 0;                            \
                                                                                 \
-                C_VAR(b, _uniq);                                                \
-        })
+                        C_VAR(b, _uniq);                                        \
+                }))
```

Comment:
This evaluates to `1 ? (!!(_x)) : ...` in case of constant-expressions, but the tertiary operator is not a constant expression in C. Did you check this works?

The biggest issue with constant-expression-checks in macros is that the `else-clause` will end up as dead-code, but still be generated and thus fail const-expr evaluations. We avoided that in `C_CC_MACRO*()` by using `__builtin_choose_expr()`.

Reply:
Btw., you can test for constant-expressions via `__builtin_constant_p()` in the test-suite, to verify this works.

Reply:
> This evaluates to 1 ? (!!(_x)) : ... in case of constant-expressions, but the tertiary operator is not a constant expression in C. Did you check this works?

It worked, in the sense that it suppressed the warning with (my version of) clang. (my version of) gcc never had a problem to begin with. I think, if all operands of "?:" are const, it's still a constant expression. But that isn't the case here. Updated to use `__builtin_choose_expr()`.

Repushed.

Reply:
You are right, I was thinking of the comma-operator, which is not a constant-expression. That's a C oddity that I never understood, but probably no-one cared enough to make it a const-expr if both operands are const-exprs.

But yeah, the "false"-path of ternary-operators has caused me lots of trouble in the past, because it still is compiled and checked, etc. Thanks for the adjustments!